includes/chamilo-functions.php

```diff
@@ -181,4 +191,236 @@ function chamilo_display_courses_list($courses) {
+  
+  // Get an instance of the WC_Customer Object from the user ID
+  $customer = new WC_Customer( $user_id );
+
+  $username     = $customer->get_username(); // Get username
+  $user_email   = $customer->get_email(); // Get account email
+  $first_name   = $customer->get_first_name();
+  $last_name    = $customer->get_last_name();
+  $display_name = $customer->get_display_name();
+ 
+  
+  // Check if the PS customer have already an account in Chamilo
+        $chamilo_params = array(
+            'original_user_id_name' => 'external_user_id',
+            //required, field name about original user id
+            'original_user_id_value' => $user_id,
+            //required, field value about original user id
+            'secret_key' => $finalKey
+            //required, secret key ("your IP address and security key from chamilo") encrypted with sha1
+        );
+  		
+  		$chamilo_user_data = chamilo_soap_call( 'registration', 'WSGetUser', $chamilo_params );
+  
+        //write_log($chamilo_user_data);
+        $chamilo_user_id = null;
+        if (!empty($chamilo_user_data)) {
+            $chamilo_user_id = $chamilo_user_data->user_id;
+        }
+  
+  
+  		// Login generation - firstname (30 len char) + PS customer id
+        
+        $login = substr(strtolower($username),0,30).$user_id;
+        // User does not have a Chamilo account we proceed to create it
+        if (empty($chamilo_user_id)) {
+            
+			//write_log( "Wordpress Customer does not exist in Chamilo proceed the creation of the Chamilo user" );
+            // Password generation
+            $password = $clean_password = generate_password();
+          	$encryption = 'sha1';
+            switch($encryption) {
+                case 'md5':
+                    $password = md5($password);
+                    break;
+                case 'sha1':
+                    $password = sha1($password);
+                    break;
+            }
+
+            // Default account validity in chamilo.
+            $expirationDate = date('Y-m-d H:i:s', strtotime("+3660 days"));
+
+            // Setting params
+            $chamilo_params =
+                array(
+                    'firstname' => $first_name,   // required
+                    'lastname' => $last_name,    // required
+                    'status' => '5',                    // required, (1 => teacher, 5 => learner)
+                    'email' => $user_email,       // optional, follow the same format (example@domain.com)
+                    'loginname' => $login,                 // required
+              		'official_code'=> $user_id,
+                    'password' => $password,              // required, it's important to define the salt into an extra field param
+                    'encrypt_method' => $encryption,  // required, check if the encrypt is the same than dokeos configuration
+                    'language' => 'brazilian',              // optional
```

Comment:
language shouldn't be a fixed value. Isn't there something we can use in WordPress to catch the current language?

Reply:
Yes @ywarnier , but not getting entire name of language, just the code. 
$lang = get_bloginfo("language");
echo $lang;

// output fr-FR, pt-BR, ....
So for this function, maybe the 'language' just let empty to chamilo made of default language in chamilo platform.